doc/devel/development_setup.rst

```diff
@@ -4,6 +4,51 @@
 Setting up Matplotlib for development
 =====================================
 
+.. note::
+
+    Setting up everything correctly and working with git can be challenging for
+    new contributors. If you have difficulties, feel free to ask for help. We
+    have a dedicated :ref:`contributor-incubator` channel on gitter for
+    onboarding new contributors.
+
+Setting up git and retrieving the latest version of the code
+============================================================
+
+Matplotlib is hosted on `GitHub <https://github.com/matplotlib/matplotlib>`_.
+To contribute you will need to sign up for a `free GitHub account
+<https://github.com/signup/free>`_.
+
+GitHub uses Git for version control, which allows many people to work together
+on the project. See the following links for more information:
+
+- `GitHub help pages <https://help.github.com/>`_
+- `NumPy documentation <https://numpy.org/doc/stable/dev/index.html>`_
+- `pandas documentation <https://pandas.pydata.org/docs/development/contributing.html#working-with-the-code>`_
+
+Forking
+-------
+Go to https://github.com/matplotlib/matplotlib and click the `Fork` button in
+the top right corner to create your own copy of the project.
+
+Now clone your fork to your machine:
+
+.. code-block:: bash
+
+    git clone https://github.com/your-user-name/matplotlib.git
+    cd matplotlib
+    git remote add upstream https://github.com/matplotlib/matplotlib.git
+
+This creates a directory `matplotlib` under your current working directory
+with the matplotlib source code. The last command connects you to the
+official matplotlib repository under the remote name `upstream`.
```

Comment:
```suggestion
This creates a directory :file:`matplotlib` under your current working directory
with the Matplotlib source code. The last command connects you to the
official Matplotlib repository under the remote name ``upstream``.
```
(or just use code style for the directory name)

Reply:
Even though technically incorrect, I've left the casing to lowercase here. Because the folder and also the git project names/urls use lower-case. Making the distinction between the project's abstract name and its naming in the file system and on github feels like creating more confusion than it's worth.
But no too strong opinion if somebody feels the Name-casing is super-important.

That said, I'd support if somebody would decide that the project name itself will become lowercase. :smile: